crates/rust-analyzer/src/bin/main.rs

```diff
@@ -21,14 +21,25 @@ static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
 
 fn main() {
     if let Err(err) = try_main() {
+        log::error!("Unexpected error: {}", err);
         eprintln!("{}", err);
         process::exit(101);
     }
 }
 
 fn try_main() -> Result<()> {
     let args = args::Args::parse()?;
-    setup_logging(args.log_file)?;
+
+    #[cfg(debug_assertions)]
+    if args.wait_dbg || env::var("RA_WAIT_DBG").is_ok() {
+        #[allow(unused_mut)]
+        let mut d = 4;
+        while d == 4 {
+            d = 4;
+        }
```

Comment:
What about an `#[inline(never)]` function with an infinite loop (`loop {}`) from which you can forcefully return? That should also work with optimizations I think.

Reply:
The code has no sense for release builds, so it's great if the optimizer can throw it away automatically :)